### PR TITLE
[Experimental] Price Slider block: add colors panel

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

## ~⚠️  Branched off 52506~

~Please, review https://github.com/woocommerce/woocommerce/pull/52506 first.
I want to make PR shorter when it's possible.~

### Changes proposed in this Pull Request:

This PR adds the `colors` panel (in the block sidebar) to the Price Slider block.

Part of https://github.com/woocommerce/woocommerce/issues/51821
Closes 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' build` command to have experimental blocks available.
1. Ensure to add a few products to your store
2. Ensure to rating products
3. Add the **Product Filters (Experimental)** block to the Product Catalog template.
4. Focus on the Price Slider block.

image

6. Ensure block sidebar is opened/visible
7. Focus the sidebar styles tab
8. Confirm you see the `colors` panel

image

9. Set different colors
10. Confirm the slider changes its colors
11. Save the post.
12. Hard refresh
13. Confirm the colors are currently applied in the front- end.

https://github.com/user-attachments/assets/ff02140e-e6d2-4f50-9408-08483ca57baf

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx [hunk 1]
```diff
@@ -3,25 +3,70 @@
...
+			colorNames
+		),
 	} );
 
-	const { isLoading, price } = context.filterData;
+	const colorGradientSettings = useMultipleOriginColorsAndGradients();
 
 	if ( isLoading ) {
-		return 'Loading...';
+		return <>{ __( 'Loading…', 'woocommerce' ) }</>;
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx:69]: Nice catch!

**Review by thealexandrelara — COMMENTED** [review]
Great work, thank you for working on this.

Here are the testing results:

✅ Scenario 1: Merchants can edit the slider colors
- Conditions:
    - A merchant visits the Product Catalog template and adds the Product Filters (Experimental) block.
    - The merchant selects the Price Slider block and edit the "Slider Handle", "Slider Handle Border", and the "Slider" settings.
- Expected Outcome: The selected colors for each setting is correctly displayed in the canvas.
- Details: The selected colors are correctly displayed within the editor.

---

✅ Scenario 2: Shoppers can see the customized slider colors
- Conditions:
    - A merchant visits the Product Catalog template and adds the Product Filters (Experimental) block.
    - The merchant selects the Price Slider block and edit the "Slider Handle", "Slider Handle Border", and the "Slider" settings.
    - A shopper visits the Shop page.
- Expected Outcome: The shopper sees the colors for the price slider exactly as customized by the merchant.
- Details: The colors are correctly displayed on the frontend.

---

I left I question in the original issue https://github.com/woocommerce/woocommerce/issues/51821 regarding the customization of the price slider thumb. Basically, I noticed that when you hover the mouse over the thumb, the color is set back to the default color, is that expected?

https://github.com/user-attachments/assets/16373753-5a38-4f3f-b0ad-fa318fadd8c5

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx:69]: :-D anyway I think this is something we'd like to replace with a better _loading_ implementation

**Comment by retrofox** [comment]
> Great work, thank you for working on this.

🙇‍♂️
 
> I left I question in the original issue #51821 regarding the customization of the price slider thumb. Basically, I noticed that when you hover the mouse over the thumb, the color is set back to the default color, is that expected?

Since it's an experimental block, I wonder whether we can merge this PR and continue the implementation in follow-ups.
I like this approach because it helps to avoid huge PRs, making the dev process as well as the review process easier.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx (hunk 1 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx:69]: > anyway I think this is something we'd like to replace with a better loading implementation

We should. I used the text just as a POC while building the slider but forgot to replace it with a proper loading animation.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Order of colors in the array
+ * affects the order of CSS variables and classes
+ * in the generated CSS.
+ * Changing the order will break the markup
+ * generated by the block,
+ * producing validation errors.
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts]: Nitpick: Can we use content width here, for example, 80 characters? The line breaks seem random to me.

**Review by dinhtungdu — APPROVED** [review]
> I like this approach because it helps to avoid huge PRs, making the dev process as well as the review process easier.

I generally agree with smaller PRs. Just want to share my 2 cents for striking a balance, as reviewing a small PR is easy, but reviewing multiple small PRs is not because of context switching. For this specific case, I would address the hover issue of the slider handle as it's quite small and doesn't make this bigger. But it's up to you, so I'm approving this PR.

Just noting that even if you address the hover styling issue, this PR can't close the connect issue as we haven't implemented some colors in the new design:

image

Note that, because we use the new structure, I don't think the background color makes sense for the slider anymore. It's always good to ask Jill for advice in cases like this IMO.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts [hunk 3]
```diff
@@ -45,3 +45,10 @@ export type FilterBlockContext = {
 		} >;
 	};
 };
+
+export type Color = {
+	slug: string;
+	class: string;
+	name: string;
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: These fields can be undefined when the color is a custom one.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts (hunk 2 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts]: I do agree with 80 charts, but especially the maximum (vm ftw)
I tried to break the lines in the proper words, but let me improve it :-D.
Thanks for he feedback

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts (hunk 3 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: Nice catch. Updating

**Comment by retrofox** [comment]
> > I like this approach because it helps to avoid huge PRs, making the dev process as well as the review process easier.
> I generally agree with smaller PRs. Just want to share my 2 cents for striking a balance, as reviewing a small PR is easy, but reviewing multiple small PRs is not because of context switching. 

Fair enough. Let's improve the balance

> For this specific case, I would address the hover issue of the slider handle as it's quite small and doesn't make this bigger. 

I agree, too. In my defense, if I'm not wrong, the hover color was aded just a few hours ago, isn't it?


> Just noting that even if you address the hover styling issue, this PR can't close the connect issue as we haven't implemented some colors in the new design:

Sorry, I'm not following here. Could you clarify?

> Note that, because we use the new structure, I don't think the background color makes sense for the slider anymore. It's always good to ask Jill for advice in cases like this IMO.

cc @j111q

**Comment by dinhtungdu** [comment]
> I agree, too. In my defense, if I'm not wrong, the hover color was aded just a few hours ago, isn't it?

@retrofox I don't think so, I just based on the Alexandre review and your reply to him. We don't have a hover style in the design. For this PR, we can simply set the thumb style always the same. We can improve it later in followup PRs. The issue I'm referring to is [here](https://github.com/woocommerce/woocommerce/pull/52507#issuecomment-2455541144).

> > Just noting that even if you address the hover styling issue, this PR can't close the connect issue as we haven't implemented some colors in the new design:

> Sorry, I'm not following here. Could you clarify?

I mean that, in the design, we have style settings for text fields, which is a part of the slider. So it should belong to https://github.com/woocommerce/woocommerce/issues/51821. This PR doesn't add color settings for text fields, so it can't close https://github.com/woocommerce/woocommerce/issues/51821 as its description states.

**Comment by retrofox** [comment]
> > I agree, too. In my defense, if I'm not wrong, the hover color was aded just a few hours ago, isn't it?
> 
> @retrofox I don't think so, I just based on the Alexandre review and your reply to him. We don't have a hover style in the design. For this PR, we can simply set the thumb style always the same. We can improve it later in followup PRs. The issue I'm referring to is [here](https://github.com/woocommerce/woocommerce/pull/52507#issuecomment-2455541144).

👍 
> 
> > > Just noting that even if you address the hover styling issue, this PR can't close the connect issue as we haven't implemented some colors in the new design:
> 
> > Sorry, I'm not following here. Could you clarify?
> 
> I mean that, in the design, we have style settings for text fields, which is a part of the slider. So it should belong to #51821. This PR doesn't add color settings for text fields, so it can't close #51821 as its description states.

Gotcha. Edited the PR replacing `Closes` with `Part of`

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts (hunk 2 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/constants.ts]: addressed here f9dd722ce5a531589804756912c489820cb37c4b

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts (hunk 3 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: 779b0d5627443adfd65624622f80d652f45d1466

**Comment by retrofox** [comment]
Going to merge soon. In case you'd like to take a look, especially at the code that[ applies background color with transparency to the slider sides](https://github.com/woocommerce/woocommerce/pull/52507/commits/45106fd16c51c1936e3374c45bc410f4866d676a).

**Pull request merged**